pkg/controller/factory/factory.go

```diff
@@ -82,6 +92,18 @@ func (f *Factory) WithInformers(informers ...Informer) *Factory {
 	return f
 }
 
+// WithFilteredEventsInformers is used to register event handlers and get the caches synchronized functions.
+// Pass the informers you want to use to react to changes on resources. If informer event is observed, then the Sync() function
+// is called.
+// Pass filter to filter out events that should not trigger Sync() call.
+func (f *Factory) WithFilteredEventsInformers(filter EventFilterFunc, informers ...Informer) *Factory {
+	f.filteredInformers = append(f.filteredInformers, filteredInformers{
```

Comment:
having two informer lists is looking pretty strange. Can't we unify that?

Reply:
sure can, I'll merge those